src/hud.cpp

```diff
@@ -196,24 +196,25 @@ void Hud::drawItem(const ItemStack &item, const core::rect<s32>& rect,
 	}
 
 //NOTE: selectitem = 0 -> no selected; selectitem 1-based
-void Hud::drawItems(v2s32 upperleftpos, s32 itemcount, s32 offset,
-		InventoryList *mainlist, u16 selectitem, u16 direction)
+void Hud::drawItems(v2s32 upperleftpos, s32 itemcount, s32 inv_offset,
+		InventoryList *mainlist, u16 selectitem, u16 direction, const v2s32 &offset)
```

Comment:
Why is offset a const ref while upperleftpos isn't?  Please make both a const ref, or pass offset by value.